### ISSUE TITLE
No errors when trying to set credentials without backends

### ISSUE THREAD

**Issue by lancewl** [jaraco/keyring#541]
**Describe the bug**
No errors when trying to set credentials without any backends there.

**To Reproduce**
```
>>> import keyring
>>> keyring.get_keyring()
<keyring.backends.null.Keyring object at 0x7f62e2207c50>
>>> keyring.set_password("system", "username", "password")
```

**Expected behavior**
Shouldn't the set_password throw the NoKeyringError since there's not backend there? I'm trying to disable keyring in my environment. I expect that it will throw NoKeyringError so that I can catch it and deal with the situation that Keyring doesn't work. But it just keep silent when trying to set credentials without proper backend.

**Environment**

- OS: Ubuntu 20.04.3 LTS

```
$ pip list | grep keyring
keyring            23.2.1
$ keyring --list-backends
keyring.backends.chainer.ChainerBackend (priority: -1)
keyring.backends.fail.Keyring (priority: 0)
```

**Additional context**
I used to use Jeepney and keyring-jeepney to setup a backend in WSL. I've tried to uninstall both of them and follow the disable instructions, but seems like Keyring is still detecting some backend.

**Comment by jaraco** [comment]
Hmm. I notice in your repro, the backend is `null.Keyring`. That backend has a priority of -1 and should never be automatically selected. That suggests to me that you ran `keyring --disable`, which is mainly meant to configure the null keyring that is mainly designed to return None for any 'get_password' operation.

If you remove the configuration file that `--disable` added, you should start getting the 'fail' keyring by default, which will raise an error if anyone attempts to set a credential. But then, get operations will also raise errors.

It sounds like maybe what you want is a backend that raises an error on set but returns None on get. Is that right?

We'll have to ponder what your expectations have for your environment and what implications supporting that would have on other environments. Can you explain more about how your environment is configured and what you're trying to accomplish in a bit more detail?

**Comment by lancewl** [comment]
Hey @jaraco, thanks for your reply! The main problem here is that I didn't realize that there are `null` and `fail` two backends. So I was expecting the `null` backend to throw the errors to me, which should be the behavior of `fail`.

I was using jeepney as the backend and I was struggling in remove that from my environment. I was trying to reset my environment to the status that doesn't have any supported backends, and I used `keyring --disable` which will actually configure the keyring to `null` keyring but not `fail` keyring.

After I cleanup all the packages that I installed for jeepney and the config file of keyring, I'm able to get the errors from `fail` backend.

Thanks again for explain the issue to me!
